### PR TITLE
Replacing mat-table-exporter library

### PR DESCRIPTION
**Pull request by xkostka2**
* Library mat-table-exporter, used to exporting displayed data from tables, was replaced by custom implementation in all tables
* Supported export formats have been reduced to CSV, however, everything is ready in case new formats were to be added

### CODE REVIEW

**Review by xkureck — CHANGES_REQUESTED** [review]
1. export of attributes table
image
As you can see in the picture its not nice if you are trying to export some attributes. The question is if there is use-case for exporting attributes and if it is in how form we want to export it.
same problem in resources table 
image


Checked tables:
attr-def-list.component.html 
blacklist-list.component.html 
destination-list.component.html 
generate-sponsored-members-dialog.component.html 
ext-sources-list.component.html 
facilities-list.component.html 
hosts-list.component.html 
members-candidates-list.component.html 
owners-list.component.html 
resources-tags-list.component.html 
security-teams-list.component.html 
services-list.component.html 
members-list.component.html 
sponsored-members-list.component.html 
users-list.component.html 
application-list-details.component.html 
applications-list.component.html 
attributes-list.component.html 
facility-select-table.component.html 
groups-list.component.html 
resources-list.component.html 

for the rest of the tables I would like to ask @HejdaJakub to give his review

File: apps/publications/src/app/components/authors-list/authors-list.component.html [hunk 1]
```diff
@@ -1,17 +1,15 @@
-<div [class.hide-table]="exporting" [hidden]="authors.length === 0 || dataSource.filteredData.length === 0" class="card mt-2">
+<div [hidden]="authors.length === 0 || dataSource.filteredData.length === 0" class="card mt-2">
   <div class="card-body">
-    <perun-web-apps-table-options (exportEnd)="exporting = false" (exportStart)="exporting = true" [exporter]="exporter"></perun-web-apps-table-options>
+    <perun-web-apps-table-options (exportData)="exportData($event)"></perun-web-apps-table-options>
```
Comment (HejdaJakub) [inline apps/publications/src/app/components/authors-list/authors-list.component.html:3]: Authors are exported without titles.
image

File: apps/publications/src/app/components/publications-list/publications-list.component.html [hunk 2]
```diff
@@ -1,18 +1,16 @@
-<div [class.hide-table]="exporting" [hidden]="dataSource.filteredData.length === 0 || publications.length === 0"
+<div [hidden]="dataSource.filteredData.length === 0 || publications.length === 0"
      class="card mt-3">
   <div class="card-body">
-    <perun-web-apps-table-options (exportEnd)="exporting = false" (exportStart)="exporting = true" [exporter]="exporter"
+    <perun-web-apps-table-options (exportData)="exportData($event)"
```
Comment (HejdaJakub) [inline apps/publications/src/app/components/publications-list/publications-list.component.html:4]: Columns `Reported by` and `Thanks` should be represented by names, but in exported CSV are only counts.
image

File: apps/user-profile/src/app/pages/settings-page/settings-authorization/settings-authentication.component.html [hunk 3]
```diff
@@ -17,10 +17,8 @@ <h1 class="page-subtitle mt-5">{{'AUTHENTICATION.MFA' | customTranslate | transl
 <!--  <mat-slide-toggle #toggle color="primary">{{'AUTHENTICATION.MFA_TOGGLE' | customTranslate | translate}}</mat-slide-toggle>-->
   <div [class.hide-table]="exporting" class="card mt-2">
     <div class="card-body">
-      <perun-web-apps-table-options (exportEnd)="exporting = false" (exportStart)="exporting = true"
-                                    [exporter]="exporter"></perun-web-apps-table-options>
       <div class="overflow-auto">
-        <table #exporter="matTableExporter" [dataSource]="dataSource" class="w-100" mat-table matTableExporter>
+        <table [dataSource]="dataSource" class="w-100" mat-table>
```
Comment (HejdaJakub) [inline apps/user-profile/src/app/pages/settings-page/settings-authorization/settings-authentication.component.html:21]: You only removed old matTbaleExporter and you didn't replace it by new solution. If it was your purpose, then just ignore this comment.

**Review by HejdaJakub — CHANGES_REQUESTED** [review]

**Review by xkureck — APPROVED** [review]
You have my approve for core functionality + tables that I have reviewed

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
All exported tables look good, but there is one last issue. A few days ago was merged branch with changes in Publication list (#657) and there is the thanks-list component, which uses an old exporter. But you deleted MatTableExporter, so my IDE throws an error and this component should be changed too.
Error

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**